includes/Modules/Analytics_4.php

```diff
@@ -1173,39 +1174,36 @@ function ( $metric_def ) {
 	 * Parses the orderby value of the data request into an array of AnalyticsData OrderBy object instances.
 	 *
 	 * @since 1.93.0
+	 * @since n.e.x.t Updated to provide support for ordering by dimensions.
 	 *
 	 * @param array|null $orderby Data request orderby value.
 	 * @return Google_Service_AnalyticsData_OrderBy[] An array of AnalyticsData OrderBy objects.
 	 */
 	protected function parse_reporting_orderby( $orderby ) {
-		if ( empty( $orderby ) || ! is_array( $orderby ) ) {
+		if ( empty( $orderby ) || ! is_array( $orderby ) || ! wp_is_numeric_array( $orderby ) ) {
 			return array();
 		}
```

Comment:
It is still possible to use just a single object as orderBy for a report in javascript. With this change, it will be ignored. We should not return an empty array if the `$orderby` variable is a single order object, we should convert it to an array of objects:

```php
if ( empty( $orderby ) || ! is_array( $orderby ) ) {
    return array();
}

if ( ! wp_is_numeric_array( $orderby ) ) {
    $orderby = array( $orderby );
}
```